cpp/include/proton/map.hpp

```diff
@@ -79,6 +81,12 @@ class PN_CPP_CLASS_EXTERN map {
     /// Copy a map.
     PN_CPP_EXTERN map& operator=(const map&);
 
+    /// Copy a std::map.
+    PN_CPP_EXTERN map(const std::map<K, T>&);
```

Comment:
I think this is too specific: The assigment operator uses a template to take any argument that is assignable to a proton::value: as a map - this constructor should do the same.
The assignment operator uses enable_if to ensure that it is the template type will work - I'm not sure you can use it in exactly the same way for a constructor. Actually I think we should replace the custom enable_if with std::enable_if now that we can use C++ 11